src/entity/SourcePostModeration.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  ManyToOne,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { Post } from './posts';
+import { Source } from './Source';
+import { User } from './user';
+
+export enum SourcePostModerationStatus {
+  Approved = 'approved',
+  Rejected = 'rejected',
+  Pending = 'pending',
+}
+
+@Entity()
+export class SourcePostModeration {
+  @PrimaryColumn({ type: 'text' })
+  id: string;
+
+  @Column({ type: 'text' })
+  sourceId: string;
+
+  @ManyToOne('Source', (source: Source) => source.id, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  source: Promise<Source>;
+
+  @Column({ type: 'text' })
+  status: SourcePostModerationStatus;
+
+  @Column({ type: 'text', nullable: true })
+  createdById: string | null;
+
+  @ManyToOne('User', (user: User) => user.id, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  createdBy: Promise<User>;
+
+  @Column({ type: 'text', nullable: true })
+  moderatedById: string | null;
+
+  @ManyToOne('User', (user: User) => user.id, {
+    lazy: true,
+    onDelete: 'SET NULL',
+  })
+  moderatedBy: Promise<User>;
+
+  @Column({ type: 'text', nullable: true })
+  moderatorMessage: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  rejectionReason: string | null;
+
+  @CreateDateColumn()
+  createdAt: Date;
+
+  @UpdateDateColumn()
+  updatedAt: Date;
+
+  @Column({ type: 'text', nullable: true })
+  postId: string | null;
+
+  @ManyToOne('Post', (post: Post) => post.id, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  post: Promise<Post>;
+
+  @Column({ type: 'text' })
+  type: string;
+
+  @Column({ type: 'text', nullable: true })
+  title?: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  titleHtml?: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  content?: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  contentHtml: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  image: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  sharedPostId?: string | null;
+
+  @ManyToOne('Post', (post: Post) => post.id, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  sharedPost: Promise<Post>;
+
+  @Column({ type: 'text', nullable: true })
+  url?: string | null;
```

Comment:
Added properties "url" and "imageUrl".

When users upload an external link, the image is already an url, and not a file upload. We also need to save the link to the item they are sharing since we don't have an existing post to reference to.